### ISSUE TITLE
[Bug] XTTSv2 does not produce Korean phonemes with language='ko'

### ISSUE THREAD

**Issue by ok70xx** [coqui-ai/TTS#4386]
### Describe the bug

    When using the `tts_models/multilingual/multi-dataset/xtts_v2` model with `language='ko'`, the generated audio does not sound like Korean. Some parts may have a Korean-like intonation, but the phonemes are incorrect, making it unintelligible. The problem persists even with a high-quality speaker reference file and different `speed`/`temperature` settings.    

### To Reproduce

1.  Place a clean, 14-second WAV file of a native female Korean speaker at `speaker_ref.wav`.
    9 2.  Run the following Python script in the same directory.
   10
   11 **`tts_test.py` script:**
  import torch
  from TTS.api import TTS
  import os

  --- Imports for PyTorch 2.6+ Security Fix ---
  from torch import serialization as torch_serialization
  from TTS.tts.configs.xtts_config import XttsConfig
  from TTS.tts.models.xtts import XttsAudioConfig, XttsArgs
  from TTS.config.shared_configs import BaseDatasetConfig
  ----------------------------------------------

  --- Configuration ---
  MODEL_NAME = "tts_models/multilingual/multi-dataset/xtts_v2"
  SPEAKER_WAV_PATH = "speaker_ref.wav"
  TEXT_TO_SYNTHESIZE = "상담사를 연결해 드리겠습니다. 자동주문 전화입니다. 이용중에 이전 단계를 원하시면 별표를 눌러주세요."
  LANGUAGE = "ko"
  OUTPUT_FILE = "test_output.wav"
  --- Parameters for tuning ---
  SPEED = 0.9
  TEMPERATURE = 0.6

  def normalize_korean_text(text):
      """A simplified version of the Korean cleaner for testing."""
      import re
      number_to_kor = {
          "0": "영", "1": "일", "2": "이", "3": "삼", "4": "사",
          "5": "오", "6": "육", "7": "칠", "8": "팔", "9": "구",
      }
      upper_alphabet_to_kor = {
          "A": "에이", "B": "비", "C": "씨", "D": "디", "E": "이", "F": "에프", "G": "지", "H": "에이치", "I": "아이", "J": "제이",
          "K": "케이", "L": "엘", "M": "엠", "N": "엔", "O": "오", "P": "피", "Q": "큐", "R": "알", "S": "에스", "T": "티",
          "U": "유", "V": "브이", "W": "더블유", "X": "엑스", "Y": "와이", "Z": "지",
      }
      text = text.strip()
      text = "".join(number_to_kor.get(char, char) for char in text)
      text = re.sub("[a-zA-Z]+", lambda x: "".join(upper_alphabet_to_kor.get(c, c) for c in x.group().upper()), text)
      return text

  def main():
      print("--- Coqui TTS Korean Language Test ---")
      if not os.path.exists(SPEAKER_WAV_PATH):
          print(f"[ERROR] Speaker reference file not found at: {SPEAKER_WAV_PATH}")
          return
      print(f"Found speaker file: {SPEAKER_WAV_PATH}")
      device = "cuda" if torch.cuda.is_available() else "cpu"
      print(f"Using device: {device}")
      try:
          print("Applying PyTorch 2.6+ security fix...")
          torch_serialization.add_safe_globals([XttsConfig, XttsAudioConfig, BaseDatasetConfig, XttsArgs])
          print("Security fix applied.")
          print(f"Loading model: {MODEL_NAME}...")
          tts = TTS(MODEL_NAME).to(device)
          print("Model loaded successfully.")
          cleaned_text = normalize_korean_text(TEXT_TO_SYNTHESIZE)
          print(f"Original text: '{TEXT_TO_SYNTHESIZE}'")
          print(f"Cleaned text:  '{cleaned_text}'")
          print(f"Synthesizing speech with SPEED={SPEED} and TEMPERATURE={TEMPERATURE}...")
          tts.tts_to_file(
              text=cleaned_text,
              speaker_wav=SPEAKER_WAV_PATH,
              language=LANGUAGE,
              file_path=OUTPUT_FILE,
              speed=SPEED,
              temperature=TEMPERATURE,
          )
          if os.path.exists(OUTPUT_FILE) and os.path.getsize(OUTPUT_FILE) > 0:
              print(f"\n[SUCCESS] Speech synthesized successfully and saved to '{OUTPUT_FILE}'")
          else:
              print(f"\n[ERROR] Synthesis failed. Output file was not created or is empty.")
      except Exception as e:
          print(f"\n[ERROR] An unexpected error occurred: {e}")
          import traceback
          traceback.print_exc()

  if __name__ == "__main__":
      main()

### Expected behavior

The output file `test_output.wav` should contain intelligible Korean speech.

### Logs

```shell

```

### Environment

```shell
*   **TTS version**: (Please run `pip show tts` and paste the version here)
     *   **PyTorch version**: (Please run `pip show torch` and paste the version here)
     *   **OS**: Linux
     *   **Requirements**:
         
        fastapi
        uvicorn[standard]
        python-multipart
        openai-whisper
        TTS
```

**Comment by eginhard** [comment]
The output of `tts --model_name xtts_v2 --language_idx ko --speaker_idx "Ana Florence" --text "상담사를 연결해 드리겠습니다"` sounds fine to me. Try removing the period at the end of sentences to remove hallucinations, which are common with XTTS.

**Comment by eginhard** [comment]
I'd also recommend to install our fork instead, which has many fixes and is compatible with recent versions of Python and Pytorch: https://coqui-tts.readthedocs.io/en/latest/faq.html#what-is-the-coqui-fork-about-and-how-to-install-it
